src/util/ActiveShapeUtils.tsx

```diff
@@ -45,6 +46,8 @@ function ShapeSelector<ShapePropsType>({
       return <Trapezoid {...elementProps} />;
     case 'sector':
       return <Sector {...elementProps} />;
+    case 'symbols':
+      return <Symbols {...(elementProps as unknown as SymbolsProps)} />;
```

Comment:
wish we didn't have to cast here

Reply:
+1

Reply:
Agreed. I fell back to the cast after initially not being able to resolve the type error.

This is error:
<img width="702" alt="Screenshot 2023-10-12 at 7 47 37 AM" src="https://github.com/recharts/recharts/assets/24420033/75f1f5ef-8e48-448b-b5d6-be0bee00a487">

It goes away if I add a guard:

```typescript
function isSymbolsProps(shapeType: ShapeType, _elementProps: unknown): _elementProps is SymbolsProps {
  return shapeType === 'symbols';
}
```

And then wrap the render in an if statement. It will always be true because we're already inside the `case 'symbols':` block.
```typescript
function ShapeSelector<ShapePropsType>({
  shapeType,
  elementProps,
}: {
  shapeType: ShapeType;
  elementProps: ShapePropsType;
}) {
  switch (shapeType) {
    ...
    case 'symbols':
      if (isSymbolsProps(shapeType, elementProps)) {
        return <Symbols {...elementProps} />;
      }
      break;
  }
}
```

Reply:
@ckifer @PavelVanecek  I've commited the above, but I don't mind reverting if either of you think this is not an optimal solution to the initial type error.